docs/reference/analysis/tokenfilters/synonym-graph-tokenfilter.asciidoc

```diff
@@ -50,7 +50,44 @@ PUT /test_index
 The above configures a `search_synonyms` filter, with a path of
 `analysis/synonym.txt` (relative to the `config` location). The
 `search_synonyms` analyzer is then configured with the filter.
-Additional settings are: `expand` (defaults to `true`).
+
+Additional settings are:
+
+* `expand` (defaults to `true`).
+* `lenient` (defaults to `false`). If `true` ignores exceptions while parsing the synonym configuration. It is important
+to note that only those synonym rules which cannot get parsed are ignored. For instance consider the following request:
+
+[source,js]
+--------------------------------------------------
+PUT /test_index
+{
+    "settings": {
+        "index" : {
+            "analysis" : {
+                "analyzer" : {
+                    "synonym" : {
+                        "tokenizer" : "standard",
+                        "filter" : ["my_stop", "synonym_graph"]
+                    }
+                },
+                "filter" : {
+                	"my_stop": {
+                		"type" : "stop",
+                		"stopwords": ["bar"]
+                	},
+                    "synonym_graph" : {
+                        "type" : "synonym_graph",
+                        "lenient": true,
+                        "synonyms" : ["foo, bar => baz"]
```

Comment:
However, if it was `"foo, baz => bar"` or `"bar, foo, baz"` no mapping would be added. Is this acceptable?

Reply:
@sohaibiftikhar Yes, I think it is acceptable, as long as we state it in the documentaiton

Reply:
In that case, the parser maps everything to the first word. Similarly for `WordnetParser` if you have something like
```
s(100000001,1,'bar',v,1,0).
s(100000001,2,'foo',v,1,0).
s(100000001,3,'baz',v,1,0).
```
everything gets mapped to `bar`. But since that is a stop word no mappings get added.

Reply:
> In that case, the parser maps everything to the first word.

This is only true if `expand` is explicitly set to `false`. By default, all combinations are generated, see this recreation for instance:

```
PUT test 
{
  "settings": {
    "analysis": {
      "filter": {
        "my_syns": {
          "type": "synonym_graph",
          "synonyms": [ "foo, bar, baz" ]
        }
      },
      "analyzer": {
        "my_syn_analyzer": {
          "tokenizer": "whitespace",
          "filter": [ "my_syns" ]
        }
      }
    }
  }
}

GET test/_analyze
{
  "analyzer": "my_syn_analyzer",
  "text": "a bar"
}

// Returns:

{
  "tokens": [
    {
      "token": "a",
      "start_offset": 0,
      "end_offset": 1,
      "type": "word",
      "position": 0
    },
    {
      "token": "foo",
      "start_offset": 2,
      "end_offset": 5,
      "type": "SYNONYM",
      "position": 1
    },
    {
      "token": "baz",
      "start_offset": 2,
      "end_offset": 5,
      "type": "SYNONYM",
      "position": 1
    },
    {
      "token": "bar",
      "start_offset": 2,
      "end_offset": 5,
      "type": "word",
      "position": 1
    }
  ]
}
```

Reply:
Yes. I forgot to mention the `expand` part in the documentation. Should I add that or remove that example altogether?